src/i2c.rs

```diff
@@ -803,7 +803,18 @@ where
             .write(|w| unsafe { w.bits(u32::from(addr) << 1) });
 
         // Wait until address was sent
-        while self.i2c.sr1.read().addr().bit_is_clear() {}
+        while {
+            let sr1 = self.i2c.sr1.read();
+
+            // If we received a NACK, then this is an error
+            if sr1.af().bit_is_set() {
+                self.i2c.sr1.modify(|_, w| w.af().clear_bit());
```

Comment:
Would it make sense to clear all errors here instead of just the NACK condition?

Reply:
This should now be implemented in a `check_and_clear_errors()` function.